consensus/src/chained_bft/persistent_storage.rs

```diff
@@ -234,6 +243,17 @@ impl<T: Payload> RecoveryData<T> {
             .find(|qc| qc.committed_block_id() == Some(root_block.id()))
             .ok_or_else(|| format_err!("No LI found for root: {}", root_id))?
             .clone();
+
+        ensure!(
+            storage_ledger.timestamp_usecs()
+                <= root_ledger_info
+                    .ledger_info()
+                    .ledger_info()
+                    .timestamp_usecs(),
+            "Storage timestamp {} is ahead of root {}",
```

Comment:
Timestamp is a good (and only) check, to guarantee that the root LI(S) <= LI(root).